tsconfig.common.json

```diff
@@ -0,0 +1,23 @@
+// https://www.typescriptlang.org/tsconfig#extends
+// The configuration from this file is loaded first, then overridden by those in the inheriting config file. But path-based compiler options (outDir, outFile, rootDir, include, exclude, files) are resolved from the config file they're found in. They cannot be extended (https://github.com/microsoft/TypeScript/issues/29172).
+// Config ref: https://www.typescriptlang.org/docs/handbook/tsconfig-json.html
+{
+  // https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping
+  "extends": "@tsconfig/node14/tsconfig.json",
+  "compilerOptions": {
+    "composite": true,
+    "lib": [
+      "es2020",
```

Comment:
@AnanyaJha Is there a minimum VSCode version that we support?
@gbockus-sf Brought up this interesting question today. 

Looking at the current code we have vscode 1.49 and es target of `es6` (ES2015)

```json
  "engines": {
    "vscode": "^1.49.3"
  },
```

Looks like [1.49 uses Node 12 (Visual Studio Code August 2020)](https://code.visualstudio.com/updates/v1_49) and Node 12 would translate to [ES2019](https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping#node-12)

> This is a major Electron release and comes with Chromium 83.0.4103.122 and Node.js 12.14.1.

1. Should we update the es target to match Node 12 / VSCode 1.49 which would be ES2019 ? OR 
2. Should we update the vscode engine version to ^1.66 and the es target to match corresponding Node 14 ?

Reply:
Just remembered .. We recently switched from using Node 12 to Node 14 as Node 12 reaches EOL on 2022-04-30. Not sure if VSCode has any deprecation policy in terms of its releases.. But can we maybe switch our vscode version to the minimum vscode version that supports Node 14? - which looks like VSCode version 1.56.x (https://stackoverflow.com/a/65191145)

Reply:
Changing the target to anything up of es6 breaks the build. Think we can leave it as es6 for now and revisit upgrading the vscode version and es target in future work.

Reply:
nice digging @mohanraj-r 👍  .  Would you mind creating an associated ticket to address the integration test deps on es6?

Reply:
Created WI and added to comment